homeassistant/components/tuya/cover.py

```diff
@@ -319,15 +319,15 @@ def open_cover(self, **kwargs: Any) -> None:
             commands.append(
                 {
                     "code": self.entity_description.set_position,
-                    "value": 0,
+                    "value": 100,
```

Comment:
This will be a never-ending game.

Please provide information on why the current is incorrect (documentation resources).

Reply:
For product category `mc` (door and window controllers), I found [this](https://developer.tuya.com/en/docs/iot/s?id=K9gf48r5zjsy9) documentation that states:
<img width="574" alt="image" src="https://user-images.githubusercontent.com/4990645/145434663-95d3d59b-ce62-43c5-ab93-d4528cfdeef9.png">

Making the somewhat reasonable assumption that this is used consistently, percentage should always mean percentage open, so `open_cover` should set 100, and `close_cover` should set 0. This coincides with the docstring, and is perhaps the source of the docstring.

Reply:
That docstring... is about HA, not Tuya...

Reply:
So from the Home Assistant docs:

![image](https://user-images.githubusercontent.com/195327/145542664-d341d0d9-e78f-4b1f-8508-c145dddbb593.png)

Looking at the original draft from Tuya themselves:

![image](https://user-images.githubusercontent.com/195327/145542709-5dcc3922-96e4-4032-b719-b7d89d8baa49.png)
<https://github.com/tuya/tuya-home-assistant/blob/tuya-v2-backup/custom_components/tuya_v2/cover.py>

This is indeed flipped. This is also the case in our own `current_cover_tilt_position`. That said, we cannot simply send `100` or `0` actually, as the value may need to be scaled, so this is actually incorrect already.

Reply:
This will map Home Assistant's value, back the value is should have with Tuya

```suggestion
                    "value": round(
                        self._set_position_type.remap_value_from(
                            100, 0, 100, reverse=True
                        ),
                    ),
```

Reply:
> That docstring... is about HA, not Tuya...
> And above all, refers to angle, not position.

Oops...

Thanks for the huge assist!